fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LucenOnlineIndexingTest.java

```diff
@@ -0,0 +1,631 @@
+        try (final FDBRecordContext context = openContext()) {
+            openRecordStore(context, hook);
+            try (OnlineIndexer indexBuilder = OnlineIndexer.newBuilder()
+                    .setRecordStore(recordStore)
+                    .setIndex(index)
+                    .setIndexingPolicy(OnlineIndexer.IndexingPolicy.newBuilder()
+                            .setDeferLuceneMergeDuringIndexing(true)
+                            .build())
+                    .build()) {
+                assertTrue(recordStore.isIndexDisabled(index));
+                indexBuilder.buildIndex(true);
+            }
+        }
+        // .. and assert readable mode
+        try (final FDBRecordContext context = openContext()) {
+            openRecordStore(context, hook);
+            assertTrue(recordStore.isIndexReadable(index));
+        }
+    }
+
+    @SuppressWarnings("checkstyle:VariableDeclarationUsageDistance")
+    @ParameterizedTest
+    @ValueSource(ints = {1, 2, 3})
+    void luceneOnlineIndexingTestGroupingKeysBackgroundMerge(int groupingCount) {
+        final int groupedCount = 4 - groupingCount;
+        Index index = new Index(
+                "Map_with_auto_complete$entry-value",
+                new GroupingKeyExpression(field("entry",
+                        KeyExpression.FanType.FanOut).nest(concat(LuceneIndexTest.keys)), groupedCount),
+                LuceneIndexTypes.LUCENE,
+                ImmutableMap.of());
+
+        RecordMetaDataHook hook = metaDataBuilder -> {
+            metaDataBuilder.removeIndex(TextIndexTestUtils.SIMPLE_DEFAULT_NAME);
+            TextIndexTestUtils.addRecordTypePrefix(metaDataBuilder);
+            metaDataBuilder.addIndex(MAP_DOC, index);
+        };
+        int group = 3;
+
+        // write/overwrite records
+        boolean needMerge = false;
+        for (int iLast = 60; iLast > 40; iLast --) {
+            try (FDBRecordContext context = openContext()) {
+                openRecordStore(context, hook);
+                // recordStore.markIndexReadable(index);
+                recordStore.getIndexDeferredMaintenancePolicy().setAutoMergeDuringCommit(false);
+                for (int i = 0; i < iLast; i++) {
+                    recordStore.saveRecord(multiEntryMapDoc(77L * i, ENGINEER_JOKE + iLast, group));
+                }
+                if (recordStore.getIndexDeferredMaintenancePolicy().isMergeRequired()) {
+                    needMerge = true;
+                }
+                commit(context);
+            }
+        }
+        assertTrue(needMerge);
+
+        Tuple tuple = Tuple.from("Text 2, and 1", "Text 3, plus 1", "I am text 4, with 1");
+        String[] allFiles = listFiles(index, tuple, groupingCount);
+        int oldLength = allFiles.length;
+
+        recordStore.getIndexDeferredMaintenancePolicy().setAutoMergeDuringCommit(false);
+        try (OnlineIndexer indexBuilder = OnlineIndexer.newBuilder()
+                .setRecordStore(recordStore)
```

Comment:
You shouldn't really be using the `FDBRecordStore` outside of the context it is bound to, I think, instead you should call `recordStore.toBuilder()` in the previous block, and then call `setRecordStoreBuilder()` here.

Reply:
This setup is not needed for `OnlineIndexer` operations. I've used it for my own experiments. Now removed all these "out of the blue setups" from all the tests. 
✅